<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>882</width>
    <height>569</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Spock Control</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(34, 35, 36);
color: rgb(255, 255, 255);</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>361</width>
      <height>501</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <property name="spacing">
      <number>40</number>
     </property>
     <item row="0" column="1">
      <widget class="SpockMotorStatusBox" name="frMotorStat">
       <property name="title">
        <string>Front Right</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="SpockMotorStatusBox" name="flMotorStat">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Front Left</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="SpockMotorStatusBox" name="blMotorStat">
       <property name="title">
        <string>Back Left</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="SpockMotorStatusBox" name="brMotorStat">
       <property name="title">
        <string>Back Right</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="SpockMotorStatusBox" name="armMotorStat">
       <property name="title">
        <string>Arm</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="SpockMotorStatusBox" name="bucketMotorStat">
       <property name="title">
        <string>Bucket</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="SpockDriveTrackpad" name="driveTrackpad">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>10</y>
      <width>181</width>
      <height>301</height>
     </rect>
    </property>
    <property name="title">
     <string>Drive Trackpad</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>330</y>
      <width>171</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Arm Drive</string>
    </property>
    <widget class="SpockMotorSlider" name="armSlider">
     <property name="geometry">
      <rect>
       <x>19</x>
       <y>30</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimum">
      <number>-1023</number>
     </property>
     <property name="maximum">
      <number>1023</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLCDNumber" name="armAngleNumber">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="armLimitLabel">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>50</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>18</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⬆︎</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>430</y>
      <width>171</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Bucket Drive</string>
    </property>
    <widget class="SpockMotorSlider" name="bucketSlider">
     <property name="geometry">
      <rect>
       <x>19</x>
       <y>30</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimum">
      <number>-1023</number>
     </property>
     <property name="maximum">
      <number>1023</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLCDNumber" name="bucketAngleNumber">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="connectionLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>520</y>
      <width>561</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>9</y>
      <width>261</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>Forward View</string>
    </property>
    <widget class="QVideoWidget" name="forwardVideo" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>261</width>
       <height>161</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>200</y>
      <width>261</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>Reverse View</string>
    </property>
    <widget class="QVideoWidget" name="forwardVideo_2" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>261</width>
       <height>171</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>400</y>
      <width>261</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Bucket Controls</string>
    </property>
    <widget class="QRadioButton" name="driveRadio">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Drive</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/icons/PlayStation_button_S.svg</normaloff>:/icons/PlayStation_button_S.svg</iconset>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="scoopRadio">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Scoop</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/icons/PlayStation_button_X.svg</normaloff>:/icons/PlayStation_button_X.svg</iconset>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="dumpRadio">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Dump</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/icons/PlayStation_button_T.svg</normaloff>:/icons/PlayStation_button_T.svg</iconset>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLCDNumber" name="weightNumber">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>60</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="weighButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Weigh</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/icons/PlayStation_button_R1.svg</normaloff>:/icons/PlayStation_button_R1.svg</iconset>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="leftThrottleLine">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>170</y>
      <width>4</width>
      <height>16</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="rightThrottleLine">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>170</y>
      <width>4</width>
      <height>16</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>882</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SpockMotorStatusBox</class>
   <extends>QGroupBox</extends>
   <header>spockmotorstatusbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SpockDriveTrackpad</class>
   <extends>QGroupBox</extends>
   <header>spockdrivetrackpad.h</header>
   <container>1</container>
   <slots>
    <signal>_mouseMoveEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
    <signal>_mousePressEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
    <signal>_mouseReleaseEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>SpockMotorSlider</class>
   <extends>QSlider</extends>
   <header>spockmotorslider.h</header>
  </customwidget>
  <customwidget>
   <class>QVideoWidget</class>
   <extends>QWidget</extends>
   <header location="global">QVideoWidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>armSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>armSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bucketSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>bucketSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>driveTrackpad</sender>
   <signal>_mouseMoveEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
   <receiver>MainWindow</receiver>
   <slot>driveMove(SpockDriveTrackpad*,QMouseEvent*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>490</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>driveTrackpad</sender>
   <signal>_mouseReleaseEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
   <receiver>MainWindow</receiver>
   <slot>driveRelease(SpockDriveTrackpad*,QMouseEvent*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>490</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>driveTrackpad</sender>
   <signal>_mousePressEvent(SpockDriveTrackpad*,QMouseEvent*)</signal>
   <receiver>MainWindow</receiver>
   <slot>driveMove(SpockDriveTrackpad*,QMouseEvent*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>490</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>driveRadio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>bucketDrive()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>650</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scoopRadio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>bucketScoop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>730</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dumpRadio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>bucketDump()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>810</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>weighButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>bucketWeigh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>655</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>armSliderChanged(int)</slot>
  <slot>bucketSliderChanged(int)</slot>
  <slot>driveMove(SpockDriveTrackpad*,QMouseEvent*)</slot>
  <slot>driveRelease(SpockDriveTrackpad*,QMouseEvent*)</slot>
  <slot>bucketDrive()</slot>
  <slot>bucketScoop()</slot>
  <slot>bucketDump()</slot>
  <slot>bucketWeigh()</slot>
 </slots>
</ui>
